modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java

```diff
@@ -1565,9 +1566,71 @@ public void testAddObjectEntryWithObjectValidationRule() throws Exception {
 
 		_assertCount(5);
 
-		// Names must be equals
+		// Must match user locale
 
 		ObjectValidationRule objectValidationRule6 = _addObjectValidationRule(
+			ObjectValidationRuleConstants.ENGINE_TYPE_GROOVY,
+			HashMapBuilder.put(
+				LocaleUtil.CHINA, RandomTestUtil.randomString()
+			).put(
+				LocaleUtil.US, RandomTestUtil.randomString()
+			).build(),
+			"invalidFields = true;");
+
+		User user = TestPropsValues.getUser();
+
+		user = _userLocalService.updateLanguageId(
+			user.getUserId(), LanguageUtil.getLanguageId(LocaleUtil.CHINA));
+
+		values = HashMapBuilder.<String, Serializable>put(
+			"birthday", "2000-12-25"
+		).put(
+			"date", tomorrowLocalDate.toString()
+		).put(
+			"emailAddressRequired", "bob@liferay.com"
+		).put(
+			"listTypeEntryKeyRequired", "listTypeEntryKey1"
+		).put(
+			"time", timeString
+		).build();
+
+		try {
+			_addObjectEntry(values);
+
+			Assert.fail();
+		}
+		catch (ModelListenerException modelListenerException) {
+			ObjectValidationRuleEngineException
+				objectValidationRuleEngineException =
+					(ObjectValidationRuleEngineException)
+						modelListenerException.getCause();
+
+			List<ObjectValidationRuleResult> objectValidationRuleResults =
+				objectValidationRuleEngineException.
+					getObjectValidationRuleResults();
+
+			Assert.assertEquals(
+				objectValidationRuleResults.toString(), 1,
+				objectValidationRuleResults.size());
+
+			_assertObjectValidationRuleResult(
+				objectValidationRule6.getErrorLabel(user.getLanguageId()), null,
+				objectValidationRuleResults.get(0));
+		}
+
+		// Disable object validation rule 6
+
+		objectValidationRule6.setActive(false);
+
+		_objectValidationRuleLocalService.updateObjectValidationRule(
+			objectValidationRule6);
+
+		_userLocalService.updateLanguageId(
+			user.getUserId(), LanguageUtil.getLanguageId(LocaleUtil.US));
+
+		// Names must be equals
+
+		ObjectValidationRule objectValidationRule7 = _addObjectValidationRule(
```

Comment:
Hey @rijema I'd create a new test method for this use case in ObjectValidatonRuleLocalServiceTest, something like this:


````java
	@Test
	public void testGetErrorLabel() throws Exception {
		// I'd create another user to avoid causing side effects in other tests
		
		User user = UserTestUtil.addOmniadminUser();

		user = _userLocalService.updateLanguageId(
			user.getUserId(), LanguageUtil.getLanguageId(LocaleUtil.BRAZIL));

		PermissionThreadLocal.setPermissionChecker(
			PermissionCheckerFactoryUtil.create(user));

		PrincipalThreadLocal.setName(user.getUserId());

		ObjectValidationRule objectValidationRule =
			_addObjectValidationRule(
				ObjectValidationRuleConstants.ENGINE_TYPE_DDM,
				HashMapBuilder.put(
					LocaleUtil.BRAZIL, RandomTestUtil.randomString()
				).put(
					LocaleUtil.US, RandomTestUtil.randomString()
				).build(),
				StringPool.BLANK,
				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
				_VALID_DDM_SCRIPT);

		try {
			_objectEntryLocalService.addObjectEntry(
				TestPropsValues.getUserId(), 0,
				_objectDefinition.getObjectDefinitionId(),
				HashMapBuilder.<String, Serializable>put(
					"textObjectField", RandomTestUtil.randomString()
				).build(),
				ServiceContextTestUtil.getServiceContext());

			Assert.fail();
		}
		catch (ModelListenerException modelListenerException) {
			// Assert the message here
		}
	}
````

I believe this use case would make sense to be isolated.

Thank you!

Reply:
Hi @MarinhoFeliphe , I just updated with the changes could you please take a look?